### PR TITLE
add -p command's h/ prefix does not run properly

### LINKED ISSUE #132: [PE-D][Tester D] Executing an `add -p` command from the example crashes the app
Running this add command based on the example:
`add -p n/SEMAKAU LANDFILL a/Straits Boulevard p/1,000 h/CONDO t/high-end`

crashes the app.

-------------
Labels: `type.FeatureFlaw` `severity.High`
original: t1mzzz/ped#5

### LINKED ISSUE #133: [PE-D][Tester C] Unable to add a new property through `add -p` command
Steps to reproduce:

1. Type `add -p n/NAME a/ADDRESS p/1000 h/CONDO` in the command box.
2. Press Enter.
3. No confirmation message is shown or the property is added.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: peppapighs/ped#3

### LINKED ISSUE #146: [PE-D][Tester B] Add -p feature, unable to add property
Unable to add property without specifying the optional parameters. I typed in according to the format provided in the UG, maybe better to clarify the UG format to include `h/` parameter as mandatory

-------------
Labels: `severity.High` `type.DocumentationBug`
original: maryjess/ped#8

### LINKED ISSUE #148: [PE-D][Tester B] Add -p feature, unable to add property when h/ parameter is given
After typing the previous issue, just discovered that the `h/PROPERTY_TYPE` is mandatory,  but not shown in UG (under format). 

Now tried to include `h/HDB` in my input, but still unable to add property.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: maryjess/ped#9

### LINKED ISSUE #174: [PE-D][Tester A] Exception `java.lang.reflect.InvocationTargetException` thrown for add commands.
I tried to run `add -p n/PINNACLE@DUXTON a/Can p/10000 h/HDB` and it threw an `InvocationTargetException` in the terminal. Unable to run the add command after copying the command from user guide and adding in `-p`. Seems to be due to how prices are handled.

* `add -p n/SEMAKAU LANDFILL a/Straits Boulevard p/1,000 h/CONDO t/high-end`

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: lesterong/ped#8

### LINKED ISSUE #185: [PE-D][Tester D] `add -p` command's `h/` prefix does not run properly
Typing the command:
`add -p n/A a/Cantonment Rd p/10000000 h/CONDO`
crashes the app. 

-------------
Labels: `type.FeatureFlaw` `severity.High`
original: t1mzzz/ped#13

### PR DESCRIPTION
**Pull request by yuehernkang**
Will default to `AVAILABLE` status if not specified in the command

Closes #185
Closes #174 
Closes #146 
Closes #148 
Closes #133
closes #132

### CODE REVIEW

**Comment by yuehernkang** [comment]
closes #185

**Comment by yuehernkang** [comment]
closes #174

**Comment by yuehernkang** [comment]
closes #146

**Comment by yuehernkang** [comment]
closes #148

**Comment by yuehernkang** [comment]
closes #133

**Comment by yuehernkang** [comment]
closes #132

**Review by lkwlkww — APPROVED** [review]

File: src/main/java/seedu/condonery/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -206,7 +206,8 @@ public static PropertyTypeEnum parsePropertyType(String propertyType) throws Par
      */
     public static PropertyStatusEnum parsePropertyStatus(String propertyStatus) throws ParseException {
         requireNonNull(propertyStatus);
-        return PropertyStatusEnum.valueOf(propertyStatus);
+        return PropertyStatusEnum.valueOf(propertyStatus) == null
```
Comment (misterpuffin) [inline src/main/java/seedu/condonery/logic/parser/ParserUtil.java:209]: I think this conditional doesn't make sense. valueOf will never return a null value but instead throw IllegalArgumentException, which you will need to catch and throw a ParseException, as stated in the method signature.

You can look at the parsePropertyType function as reference for how I previously catch the error. For example, the command `add -p a/Duxton n/Duxton p/10000 h/hdb s/INVALID VALUE` will cause the application to throw an Exception.

**Review by misterpuffin — CHANGES_REQUESTED** [review]
actually can refer to my pr here! https://github.com/AY2223S1-CS2103-W14-1/tp/pull/118/files

**Review by zacchaeuschok — APPROVED** [review]
LGTM

**Pull request merged**